test/univariate/continuous/tdist.jl

```diff
@@ -3,10 +3,19 @@ using ForwardDiff
 
 using Test
 
-@testset "Type stability of `rand` (#1614)" begin
-    if VERSION >= v"1.9.0-DEV.348"
-        # randn(::BigFloat) was only added in https://github.com/JuliaLang/julia/pull/44714
-        @inferred(rand(TDist(big"1.0")))
+@testset "TDist" begin
+    @testset "Type stability of `rand` (#1614)" begin
+        if VERSION >= v"1.9.0-DEV.348"
+            # randn(::BigFloat) was only added in https://github.com/JuliaLang/julia/pull/44714
+            @inferred(rand(TDist(big"1.0")))
+        end
+        @inferred(rand(TDist(ForwardDiff.Dual(1.0))))
+
     end
-    @inferred(rand(TDist(ForwardDiff.Dual(1.0))))
+
+    @test rand(TDist(1.0)) isa Float64
+    @test rand(TDist(1.0f0)) isa Float32
+
+    @test entropy(TDist(1.0)) isa Float64
+    @test entropy(TDist(1.0f0)) isa Float32
```

Comment:
Similarly here, maybe also check type inference?
```suggestion
    for T in (Float32, Float64)
        @test @inferred(rand(TDist(T(1)))) isa T
        @test @inferred(entropy(TDist(T(1)))) isa T
    end
```